### PR TITLE
I3S-Debug-App - Add Building Scene Layer Support

### PR DESCRIPTION
**Pull request by dryabinin-actionengine**

### CODE REVIEW

File: examples/website/i3s/app-debug.js [hunk 1]
```diff
@@ -218,7 +218,8 @@ export default class App extends PureComponent {
       selectedTileId: null,
       coloredTilesMap: {},
       warnings: [],
-      appDebug: true
+      appDebug: true,
```
Comment (ibgreen) [inline examples/website/i3s/app-debug.js:221]: appDebug -> debug?

File: examples/website/i3s/app-debug.js [hunk 2]
```diff
@@ -274,7 +275,17 @@ export default class App extends PureComponent {
     this.setState({tilesetUrl, name, token});
     this.handleClearWarnings();
     const metadata = await fetch(metadataUrl).then((resp) => resp.json());
-    this.setState({metadata, tileInfo: null, normalsDebugData: []});
+    const layerUrls = await this.getLayerUrls(tilesetUrl);
+    this.setState({metadata, tileInfo: null, normalsDebugData: [], layerUrls});
+  }
+
+  async getLayerUrls(tilesetUrl) {
```
Comment (ibgreen) [inline examples/website/i3s/app-debug.js:282]: - JSDoc, typescript

File: examples/website/i3s/app-debug.js [hunk 3]
```diff
@@ -274,7 +275,17 @@ export default class App extends PureComponent {
...
+    try {
+      const tileset = await load(tilesetUrl, I3SBuildingSceneLayerLoader);
+      return tileset?.sublayers.map(sublayer => sublayer.url);
+    } catch (e) {
+      return [tilesetUrl];
```
Comment (ibgreen) [inline examples/website/i3s/app-debug.js:287]: Explain the two return value types

File: examples/website/i3s/app-debug.js [hunk 4]
```diff
@@ -483,9 +494,10 @@ export default class App extends PureComponent {
     const viewport = new WebMercatorViewport(viewState.main);
     const frustumBounds = getFrustumBounds(viewport);
 
-    return [
+    const tile3dLayers = layerUrls.map((url, index) => 
       new Tile3DLayer({
-        data: tilesetUrl,
+        id: `layer-${index}`,
```
Comment (ibgreen) [inline examples/website/i3s/app-debug.js:499]: More descriptive layer names?

**Review by ibgreen — APPROVED** [review]

File: examples/website/i3s/app-debug.js (hunk 1 above)
Comment (dryabinin-actionengine) [reply examples/website/i3s/app-debug.js:221]: renamed to `debugMode`

File: examples/website/i3s/app-debug.js (hunk 2 above)
Comment (dryabinin-actionengine) [reply examples/website/i3s/app-debug.js:282]: It is a react component file. I am not sure we should change it to typescript.
JSDoc added here https://github.com/visgl/loaders.gl/pull/1839

File: examples/website/i3s/app-debug.js (hunk 3 above)
Comment (dryabinin-actionengine) [reply examples/website/i3s/app-debug.js:287]: Explained by JSDoc here https://github.com/visgl/loaders.gl/pull/1839

File: examples/website/i3s/app-debug.js (hunk 4 above)
Comment (dryabinin-actionengine) [reply examples/website/i3s/app-debug.js:499]: Done https://github.com/visgl/loaders.gl/pull/1839

**Pull request merged**